internal/hcsoci/resources.go

```diff
@@ -34,151 +32,82 @@ type Resources struct {
-	// (bind-)mounts into a WCOW v2 Xenon.
-	vsmbMounts []string
-
-	// pipeMounts is an array of the host-paths of named pipes mounted into a utility
-	// VM.
-	pipeMounts []string
-
-	// plan9Mounts is an array of all the host paths which have been added to
-	// an LCOW utility VM
-	plan9Mounts []*uvm.Plan9Share
-
-	// netNS is the network namespace
-	netNS string
-
-	// networkEndpoints is the list of network endpoints used by the container
-	networkEndpoints []string
-
+	netNS              string
 	// createNetNS indicates if the network namespace has been created
 	createdNetNS bool
-
 	// addedNetNSToVM indicates if the network namespace has been added to the containers utility VM
 	addedNetNSToVM bool
-
-	// scsiMounts is an array of the vhd's mounted into a utility VM to support
-	// scsi device passthrough.
-	scsiMounts []scsiMount
-
-	// vpciDevices is an array of vpci device names
-	vpciDevices []string
+	// layers is a pointer to a struct of the layers paths of a container
+	layers *ImageLayers
+	// resources is an array of the resources associated with a container
+	resources []ResourceCloser
 }
 
-type scsiMount struct {
-	// path is the host path to the vhd that is mounted.
-	path string
-	// autoManage if `true` means that on cleanup, the runtime should
-	// automatically delete this vhd.
-	autoManage bool
+// ResourceCloser is a generic interface for the releasing of a resource. If a resource implements
+// this interface(which they all should), freeing of that resource should entail one call to
+// <resourceName>.Release(ctx, all)
+type ResourceCloser interface {
+	Release(context.Context) error
 }
 
-// TODO: Method on the resources?
-func ReleaseResources(ctx context.Context, r *Resources, vm *uvm.UtilityVM, all bool) error {
-	if vm != nil && r.addedNetNSToVM {
-		if err := vm.RemoveNetNS(ctx, r.netNS); err != nil {
-			log.G(ctx).Warn(err)
-		}
-		r.addedNetNSToVM = false
-	}
+// AutoManagedVHD struct representing a VHD that will be cleaned up automatically.
+type AutoManagedVHD struct {
+	hostPath string
+}
 
-	if r.createdNetNS {
-		for len(r.networkEndpoints) != 0 {
-			endpoint := r.networkEndpoints[len(r.networkEndpoints)-1]
-			err := hns.RemoveNamespaceEndpoint(r.netNS, endpoint)
-			if err != nil {
-				if !os.IsNotExist(err) {
-					return err
-				}
-				log.G(ctx).WithFields(logrus.Fields{
-					"endpointID": endpoint,
-					"netID":      r.NetNS(),
-				}).Warn("removing endpoint from namespace: does not exist")
-			}
-			r.networkEndpoints = r.networkEndpoints[:len(r.networkEndpoints)-1]
-		}
-		r.networkEndpoints = nil
-		err := hns.RemoveNamespace(r.netNS)
-		if err != nil && !os.IsNotExist(err) {
-			return err
-		}
-		r.createdNetNS = false
+// Release removes the vhd.
+func (vhd *AutoManagedVHD) Release(ctx context.Context) error {
+	if err := os.Remove(vhd.hostPath); err != nil {
+		log.G(ctx).WithError(err).Warnf("failed to remove automanage-virtual-disk at: %q", vhd.hostPath)
```

Comment:
Use `.WithField("hostPath", vhd.hostPath)` instead of inserting the VHD path into the log message.

Reply:
Chain WithError with it as well?

Reply:
Yes